engine-tests/src/test/java/org/terasology/documentation/ModuleMarkupScraper.java

```diff
@@ -0,0 +1,444 @@
+        for (Module gModule : sortedGameModules) {
+            // Add anchor tags, for example [AdditionalFruits](#AdditionalFruits)
+            moduleMapListing.append("[").append(gModule.getId()).append("](#").append(gModule.getId()).append(") ");
+        }
+        moduleMapListing.append(" | ");
+        for (Module rModule : sortedRemoteModules) {
+            moduleMapListing.append("[").append(rModule.getId()).append("](#").append(rModule.getId()).append(") ");
+        }
+        moduleMapListing.append(" | ");
+
+        StringBuilder moduleMap = new StringBuilder();
+        moduleMap.append(System.getProperty("line.separator"));
+        moduleMap.append("**Extensions:**");
+        moduleMap.append(System.getProperty("line.separator"));
+        moduleMap.append("| Installed | Remote |");
+        moduleMap.append(System.getProperty("line.separator"));
+        moduleMap.append("| -------- | ---------- |");
+        moduleMap.append(System.getProperty("line.separator"));
+        moduleMap.append(moduleMapListing);
+
+        wiki.append(moduleMap);
+
+        wiki.append(ScanModules(allSortedModules));
+        System.out.println(wiki);
+
+        String fileName = "Module_Wiki.md";
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File(fileName)));
+        writer.write(wiki.toString());
+        writer.flush();
+        writer.close();
+        System.out.println("Module Wiki generation file is ready!");
+
+        env.close();
+    }
+
+    private static String ScanModules(List<Module> allSortedModules) {
+        if (allSortedModules == null) {
+            throw new IllegalArgumentException("ModuleMarkupScraper:ScanModules() - Module must be valid.");
+        }
+
+        if (allSortedModules.isEmpty()) {
+            throw new IllegalStateException("ModuleMarkupScraper:ScanModules() - No modules found? Somethings very wrong.");
+        }
+
+        StringBuilder out = new StringBuilder();
+        DependencyResolver resolver = new DependencyResolver(moduleManager.getRegistry());
+        for (Module module : allSortedModules) {
+            Name moduleId = module.getId();
+            if (module.isCodeModule()) {
+//                out.append("## Module: ").append(moduleId);
+
+                String moduleDescription = getModuleDescription(module);
+                out.append(moduleDescription);
+
+                Set<Module> modules = new HashSet<>();
+                ResolutionResult result = resolver.resolve(moduleId);
+                if (result.isSuccess()) {
+                    modules = result.getModules();
+                } else {
+                    modules.add(module);
+                }
+
+                out.append(ExportEvents(moduleId, modules));
+            }
+            out.append(System.getProperty("line.separator"));
+        }
+        return out.toString();
+    }
+
+    private static String ExportEvents(Name moduleId, Set<Module> modules) {
```

Comment:
This is a method and thus should start with a lower case letter:
```suggestion
    private static String exportEvents(Name moduleId, Set<Module> modules) {
```

Reply:
applied lowerCamelCase